File: Segundo Trimestre/6- Ejercicios con Excepciones/java/gatoSimple/TestGatoSimple.java
package gatoSimple;

public class TestGatoSimple {
  public static void main(String[] args) {
    GatoSimple garfield = new GatoSimple("macho");
    GatoSimple tom = new GatoSimple("macho");
    GatoSimple lisa = new GatoSimple("hembra");
    GatoSimple luisa = new GatoSimple("hembra");
    System.out.println("Prueba de Apareamiento: ");
    System.out.println("Macho con Macho: ");
    try {
      GatoSimple cria1 = new GatoSimple(garfield.apareaCon(tom));
      cria1.maulla();
    } catch (ExcepcionApareamientoImposible e) {
      System.out.println(e.getMessage());
    }
    
    System.out.println("Hembra con Hembra: ");
    try {
      GatoSimple cria2 = new GatoSimple(lisa.apareaCon(luisa));
      cria2.maulla();
    } catch (ExcepcionApareamientoImposible e) {
      System.out.println(e.getMessage());
    }
    
    System.out.println("Macho con Hembra: ");
    try {
      GatoSimple cria3 = new GatoSimple(garfield.apareaCon(lisa));
      cria3.maulla();
    } catch (ExcepcionApareamientoImposible e) {
      System.out.println(e.getMessage());
    }
  }
}

File: Segundo Trimestre/6- Ejercicios con Excepciones/python/gatoSimple/TestGatoSimple.py
# -*- coding: utf-8 -*-

from GatoSimple import GatoSimple

garfield = GatoSimple("macho")
garfield.maulla()
print("Toma tarta.")
garfield.comida = "tarta selva negra"
garfield.come()

print("Toma pescado, a ver si esto te gusta.")
garfield.comida = "pescado"
garfield.come()

tom = GatoSimple("macho")
print("Tom, toma sopita de verduras.")
tom.comida = "sopa de verduras"
tom.come()

lisa = GatoSimple("hembra")
luisa = GatoSimple("hembra")

print("gatitos, a ver cómo maullais")
garfield.maulla()
tom.maulla()
lisa.maulla()

garfield.peleaCon(lisa)
lisa.peleaCon(tom)
tom.peleaCon(garfield)

print("Pruebas apareamiento: ")
print("Macho con Macho:")
cria1 = garfield.apareaCon(tom)

print("Hembra con Hembra:")
cria2 = lisa.apareaCon(luisa)

print("Macho con Hembra:")
cria3 = garfield.apareaCon(lisa)
cria3.maulla()
print(cria3.getSexo())
